crates/daphne-worker/src/storage_proxy/mod.rs

```diff
@@ -193,204 +202,208 @@ async fn storage_purge(ctx: &RequestContext<'_>) -> worker::Result<Response> {
             .await
     };
 
-    futures::try_join!(kv_delete, do_delete)?;
-    Response::empty()
+    futures::try_join!(kv_delete, do_delete)
+        .map(|_| ())
+        .map_err(|e| (StatusCode::INTERNAL_SERVER_ERROR, e.to_string()).into_response())
 }
 
-fn parse_expiration_header(ctx: &RequestContext) -> Result<Option<Time>, worker::Error> {
-    let expiration_header = ctx.req.headers().get(STORAGE_PROXY_PUT_KV_EXPIRATION)?;
-    expiration_header
-        .map(|expiration| {
-            expiration.parse::<Time>().map_err(|e| {
-                worker::Error::RustError(format!("Failed to parse expiration header: {e:?}"))
-            })
-        })
-        .transpose()
-}
+#[derive(Debug)]
+struct ExpirationHeader(Time);
 
-fn elapsed(date: &worker::Date) -> Duration {
-    Duration::from_millis(worker::Date::now().as_millis() - date.as_millis())
-}
+impl Header for ExpirationHeader {
+    fn name() -> &'static http::HeaderName {
+        static HEADER_NAME: http::HeaderName =
+            http::HeaderName::from_static(STORAGE_PROXY_PUT_KV_EXPIRATION);
 
-/// Handle a kv request.
-async fn handle_kv_request(ctx: &mut RequestContext<'_>, key: &str) -> worker::Result<Response> {
-    let start = worker::Date::now();
+        &HEADER_NAME
+    }
 
-    match ctx.req.method() {
-        worker::Method::Get => {
-            let bytes = ctx
-                .env
-                .kv(KV_BINDING_DAP_CONFIG)
-                .inspect_err(|_| {
-                    ctx.metrics
-                        .kv_request_time_seconds_observe("read", "error", elapsed(&start))
-                })?
-                .get(key)
-                .bytes()
-                .await?;
+    fn encode<E: Extend<http::HeaderValue>>(&self, values: &mut E) {
+        values.extend(http::HeaderValue::from_str(&self.0.to_string()))
+    }
+
+    fn decode<'i, I>(values: &mut I) -> Result<Self, headers::Error>
+    where
+        Self: Sized,
+        I: Iterator<Item = &'i http::HeaderValue>,
+    {
+        values
+            .next()
+            .and_then(|v| v.to_str().ok())
+            .and_then(|v| v.parse().ok())
+            .map(Self)
+            .ok_or_else(headers::Error::invalid)
+    }
+}
 
-            let elapsed = elapsed(&start);
+impl ExpirationHeader {
+    fn at_least_60s_in_the_future(self) -> Self {
```

Comment:
A little more rusty, plus more to the point
```suggestion
    fn with_kv_minimum_time_out(self) -> Self {
```

Reply:
`with_` feels like you're adding something to it, whereas my name I think conveys the idea that it's making a guarantee, almost like an assertion